### PR TITLE
feat: create edit delete post

### PR DESCRIPTION
**Pull request by shrushti2000**
Implemented create post, edit post and delete post feature

### CODE REVIEW

**Review by chanchal-panpaliya — APPROVED** [review]
1. code looks clean with proper naming .
2. looks good to merge

need to changed -> 1. The modal closed after updating/editing a post ->
image
2. remove console ->
image .

File: src/features/postSlice.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    }
+    catch(error){
+        return thunkAPI.rejectWithValue(error)
+    }
+})
+
+export const getUserPost=createAsyncThunk("post/getUserPost",async(username,thunkAPI)=>{
+    try{
+        const response=await getUserPostService(username)
+        console.log(response)
```
Comment (dev-enforced) [inline src/features/postSlice.js:26]: Can remove this debugger!!

File: src/pages/FeedPage/FeedPage.js [hunk 2]
```diff
@@ -6,20 +6,70 @@ import {
...
+  const { userPosts } = useSelector((state) => state.post);
+  const { user } = useSelector((state) => state.authentication);
+  const [showToast,setShowToast]=useState(false)
+  const dispatch = useDispatch();
+  const [postContent, setPostContent] = useState({
+    title:'',
+    content: "",
+    username: user.username,
+  });
+  console.log(userPosts);
```
Comment (dev-enforced) [inline src/pages/FeedPage/FeedPage.js:36]: Can remove this debugger!!

File: src/pages/FeedPage/FeedPage.js [hunk 3]
```diff
@@ -6,20 +6,70 @@ import {
...
+      })
+    }
+  };
+  const getSortedPosts=()=>{
+    return [...userPosts].sort(function(a,b){
+      return new Date(b['createdAt']) - new Date(a['createdAt']);
+    })
+  }
+  const sortedPosts=getSortedPosts()
+  console.log(sortedPosts)
```
Comment (dev-enforced) [inline src/pages/FeedPage/FeedPage.js:66]: Here also can remove the debugger!!

File: src/features/postSlice.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+export const addUserPost=createAsyncThunk("post/addUserPost",async(postData,thunkAPI)=>{
+    try{
+        const token=localStorage.getItem("token");
+        const response=await addPostService(postData,token)
+        return response.data.posts;
+    }catch(error){
+        return thunkAPI.rejectWithValue(error)
+    }
+})
```
Comment (dev-enforced) [inline src/features/postSlice.js:64]: Got an understanding about implementing async thunk API calls here!! Good work on this Shrushti!!

File: src/store/index.js [hunk 5]
```diff
@@ -1,8 +1,30 @@
...
+import {
+  persistStore,
+  persistReducer,
+  FLUSH,
+  REHYDRATE,
+  PAUSE,
+  PERSIST,
+  PURGE,
+  REGISTER,
+} from 'redux-persist';
```
Comment (dev-enforced) [inline src/store/index.js:18]: This looks new!! Would explore it out. Do attach a reference in this thread related to using this package

**Review by dev-enforced — APPROVED** [review]
Live preview is looking good!!  https://deploy-preview-2--stellular-toffee-e3581e.netlify.app/feedpage
Do take care of the console logs in the live preview. 
The features are also working correctly!! Got new things to learn related to implement the edit action and using modals for it!!
The code also looks very well structured!!
 Do look on Chanchal's suggestions she has given an important thing to work on in the later stages as well!!

**Review by poojakelkar — APPROVED** [review]
- The code looks Clean 
- very good UI
- Small change removes all console log and debugger.
- otherwise Good to merge.

**Pull request merged**